cmd/modelcmd/modelcommand_test.go

```diff
@@ -33,75 +33,89 @@ func (s *ModelCommandSuite) SetUpTest(c *gc.C) {
 
 	s.store = jujuclient.NewMemStore()
-	s.store.CurrentControllerName = "foo"
-	s.store.Controllers["foo"] = jujuclient.ControllerDetails{}
-	s.store.Accounts["foo"] = jujuclient.AccountDetails{
-		User: "bar", Password: "hunter2",
-	}
 }
 
 var _ = gc.Suite(&ModelCommandSuite{})
 
-func (s *ModelCommandSuite) TestGetCurrentModelNothingSet(c *gc.C) {
-	s.store.CurrentControllerName = ""
-	env, err := modelcmd.GetCurrentModel(s.store)
-	c.Assert(env, gc.Equals, "")
+var modelCommandModelTests = []struct {
+	about            string
+	args             []string
+	modelEnvVar      string
+	expectController string
+	expectModel      string
+}{{
+	about:            "explicit controller and model, long form",
+	args:             []string{"--model", "bar:noncurrentbar"},
+	expectController: "bar",
+	expectModel:      "noncurrentbar",
+}, {
+	about:            "explicit controller and model, short form",
+	args:             []string{"-m", "bar:noncurrentbar"},
+	expectController: "bar",
+	expectModel:      "noncurrentbar",
+}, {
+	about:            "implicit controller, explicit model, short form",
+	args:             []string{"-m", "explicit"},
+	expectController: "foo",
+	expectModel:      "explicit",
+}, {
+	about:            "implicit controller, explicit model, long form",
+	args:             []string{"--model", "explicit"},
+	expectController: "foo",
+	expectModel:      "explicit",
+}, {
+	about:            "explicit controller, implicit model",
+	args:             []string{"--model", "bar:"},
+	expectController: "bar",
+	expectModel:      "adminbar/currentbar",
+}, {
+	about:            "controller and model in env var",
+	modelEnvVar:      "bar:noncurrentbar",
+	expectController: "bar",
+	expectModel:      "noncurrentbar",
+}, {
+	about:            "model only in env var",
+	modelEnvVar:      "noncurrentfoo",
+	expectController: "foo",
+	expectModel:      "noncurrentfoo",
+}, {
+	about:            "controller only in env var",
+	modelEnvVar:      "bar:",
+	expectController: "bar",
+	expectModel:      "adminbar/currentbar",
+}, {
+	about:            "explicit overrides env var",
+	modelEnvVar:      "bar:noncurrentbar",
+	args:             []string{"-m", "noncurrentfoo"},
+	expectController: "foo",
+	expectModel:      "noncurrentfoo",
+}}
+
+func (s *ModelCommandSuite) TestModelName(c *gc.C) {
+	s.store.Controllers["foo"] = jujuclient.ControllerDetails{}
+	s.store.Controllers["bar"] = jujuclient.ControllerDetails{}
+	s.store.CurrentControllerName = "foo"
+	s.store.Accounts["foo"] = jujuclient.AccountDetails{
+		User: "bar", Password: "hunter2",
+	}
+	err := s.store.UpdateModel("foo", "adminfoo/currentfoo", jujuclient.ModelDetails{"uuidfoo1"})
 	c.Assert(err, jc.ErrorIsNil)
-}
-
-func (s *ModelCommandSuite) TestGetCurrentModelCurrentControllerNoCurrentModel(c *gc.C) {
-	env, err := modelcmd.GetCurrentModel(s.store)
-	c.Assert(env, gc.Equals, "foo:")
+	err = s.store.UpdateModel("foo", "adminfoo/oncurrentfoo", jujuclient.ModelDetails{"uuidfoo2"})
```

Comment:
isn't this a typo? "oncurrentfoo"
If no tests failed, maybe this line isn't needed?